### PR TITLE
Upgrading clang-tidy to 7

### PR DESCRIPTION
**Pull request by veblush**
Since clang-tidy became able to diagnose files without compile errors, it's now possible to upgrade it to 7 or later. Let's bump clang-tidy to 7, the default version on Debian "buster".

This is a follow-up of #24193

### CODE REVIEW

File: src/core/lib/security/credentials/tls/grpc_tls_credentials_options.h
Comment (markdroth) [inline src/core/lib/security/credentials/tls/grpc_tls_credentials_options.h:69]: Instead of making this a const ref, how about using `std::move()` in the body?  That way, we avoid a copy in cases where people pass an rvalue.

File: src/core/lib/transport/metadata.h
Comment (markdroth) [inline src/core/lib/transport/metadata.h:112]: Don't we need to add parens on this line as well?

File: src/cpp/server/load_reporter/load_reporter_async_service_impl.cc
Comment (markdroth) [inline src/cpp/server/load_reporter/load_reporter_async_service_impl.cc:361]: This change doesn't look right to me.  I think this function is supposed to release the last ref when it finishes.

File: BUILD
Comment (markdroth) [inline BUILD:1813]: I assume these lines are being commented out just for testing.

**Comment by markdroth** [comment]
BTW, note that when you create a PR that you don't intend to be reviewed yet, you can create it as a draft PR, which will avoid adding reviewers from codeowners.

File: BUILD
Comment (veblush) [reply BUILD:1813]: This is actually intended because those headers are not able to be compiled so it should be excluded from the build file. Previously those headers are not included by sources, it happened to pass the build tests.

File: src/core/lib/security/credentials/tls/grpc_tls_credentials_options.h
Comment (veblush) [reply src/core/lib/security/credentials/tls/grpc_tls_credentials_options.h:69]: That'd work, too.

File: src/core/lib/transport/metadata.h
Comment (veblush) [reply src/core/lib/transport/metadata.h:112]: Good catch!. Automatic fix wasn't able to fix statements in a #else block.

File: src/cpp/server/load_reporter/load_reporter_async_service_impl.cc
Comment (veblush) [reply src/cpp/server/load_reporter/load_reporter_async_service_impl.cc:361]: The one holding the last ref is `CallableTag` so technically this would work but I like to make this function consistent with other handlers in the class. So I'm rather reverting this and adding NOLINT comment to the function.

File: BUILD
Comment (markdroth) [reply BUILD:1813]: @ashithasantosh If these files can't actually be compiled, we should fix that.

File: src/core/ext/filters/client_channel/lb_policy/subchannel_list.h
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/subchannel_list.h:368]: Please also change line 366 from `const ServerAddress&` to `ServerAddress`, and change line 386 to use `std::move(address)`.

Comment (veblush) [reply src/core/ext/filters/client_channel/lb_policy/subchannel_list.h:368]: Done!

**Comment by veblush** [comment]
Thank you, Mark for the review. This PR was ready for the view but it started failing once it's rebased into the base PR. That's why I had to fix some errors after sending a review request.

File: src/core/ext/xds/certificate_provider_store.h
Comment (markdroth) [inline src/core/ext/xds/certificate_provider_store.h:30]: This include should go before the previous block, as per the style guide.

https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes

**Review by markdroth — APPROVED** [review]
This looks great, modulo one remaining comment, but I'll go ahead and approve.

File: src/core/ext/xds/certificate_provider_store.h
Comment (veblush) [reply src/core/ext/xds/certificate_provider_store.h]: Oops I missed. Thanks!

File: BUILD
Comment (ashithasantosh) [reply BUILD:1813]: Sure! Will look into it after clang-tidy update is merged.

**Review by jtattermusch — COMMENTED** [review]
LGTM for the commit that upgrades clang 6 to clang 7. Leaving remaining parts of the PR for others to review.

**Pull request merged**
